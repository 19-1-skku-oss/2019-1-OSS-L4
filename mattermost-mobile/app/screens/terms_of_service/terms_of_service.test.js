// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import TermsOfService from './terms_of_service.js';

jest.mock('react-intl');

jest.mock('app/utils/theme', () => {
    const original = require.requireActual('app/utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('TermsOfService', () => {
    const actions = {
        getTermsOfService: jest.fn(),
        updateMyTermsOfServiceStatus: jest.fn(),
        logout: jest.fn(),
    };

    const baseProps = {
        actions,
        navigator: {
            dismissAllModals: jest.fn(),
            dismissModal: jest.fn(),
            setButtons: jest.fn(),
            setOnNavigatorEvent: jest.fn(),
        },
        theme: Preferences.THEMES.default,
        closeButton: {},
        siteName: 'Mattermost',
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <TermsOfService {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for get terms', async () => {
        const wrapper = shallow(
            <TermsOfService {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        await actions.getTermsOfService();
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for fail of get terms', async () => {
        const getTermsOfService = async () => {
            return {
                error: {},
            };
        };

        const props = {
            ...baseProps,
            actions: {
                ...actions,
                getTermsOfService,
            },
        };

        const wrapper = shallow(
            <TermsOfService {...props}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.state('loading')).toEqual(true);
        await getTermsOfService();
        expect(wrapper.state('loading')).toEqual(false);
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call props.navigator.setButtons on setNavigatorButtons', async () => {
        const wrapper = shallow(
            <TermsOfService {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});

        expect(baseProps.navigator.setButtons).toHaveBeenCalledTimes(2);
        wrapper.instance().setNavigatorButtons(true);
        expect(baseProps.navigator.setButtons).toHaveBeenCalledTimes(3);
        wrapper.instance().setNavigatorButtons(false);
        expect(baseProps.navigator.setButtons).toHaveBeenCalledTimes(4);
    });

    test('should enable/disable navigator buttons on setNavigatorButtons true/false', () => {
        const wrapper = shallow(
            <TermsOfService {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});

        wrapper.instance().setNavigatorButtons(true);
        expect(wrapper.getElement()).toMatchSnapshot();
        wrapper.instance().setNavigatorButtons(false);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot on enableNavigatorLogout', () => {
        const wrapper = shallow(
            <TermsOfService {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});
        wrapper.instance().enableNavigatorLogout();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call dismissAllModals on closeTermsAndLogout', () => {
        const wrapper = shallow(
            <TermsOfService {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.setState({loading: false, termsId: 1, termsText: 'Terms Text'});
        wrapper.instance().closeTermsAndLogout();
        expect(baseProps.navigator.dismissAllModals).toHaveBeenCalledTimes(1);
    });
});
