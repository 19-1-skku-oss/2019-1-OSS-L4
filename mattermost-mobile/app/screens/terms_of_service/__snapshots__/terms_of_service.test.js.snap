// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsOfService should enable/disable navigator buttons on setNavigatorButtons true/false 1`] = `
<React.Fragment>
  <Connect(StatusBar) />
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
      }
    }
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "padding": 30,
      }
    }
  >
    <Connect(Markdown)
      baseTextStyle={
        Object {
          "color": "#3d3c40",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      blockStyles={
        Object {
          "adjacentParagraph": Object {
            "marginTop": 6,
          },
          "horizontalRule": Object {
            "backgroundColor": "#3d3c40",
            "flex": 1,
            "height": 0.5,
            "marginVertical": 10,
          },
          "quoteBlockIcon": Object {
            "color": undefined,
            "padding": 5,
          },
        }
      }
      disableAtMentions={true}
      disableChannelLink={true}
      disableHashtags={true}
      navigator={
        Object {
          "dismissAllModals": [MockFunction],
          "dismissModal": [MockFunction],
          "setButtons": [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": true,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": true,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": true,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": true,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": false,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": false,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "setOnNavigatorEvent": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        }
      }
      textStyles={
        Object {
          "code": Object {
            "alignSelf": "center",
            "backgroundColor": undefined,
            "fontFamily": "Menlo",
          },
          "codeBlock": Object {
            "fontFamily": "Menlo",
          },
          "del": Object {
            "textDecorationLine": "line-through",
          },
          "emph": Object {
            "fontStyle": "italic",
          },
          "error": Object {
            "color": "#fd5960",
          },
          "heading1": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading1Text": Object {
            "paddingBottom": 8,
          },
          "heading2": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading2Text": Object {
            "paddingBottom": 8,
          },
          "heading3": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading3Text": Object {
            "paddingBottom": 8,
          },
          "heading4": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading4Text": Object {
            "paddingBottom": 8,
          },
          "heading5": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading5Text": Object {
            "paddingBottom": 8,
          },
          "heading6": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading6Text": Object {
            "paddingBottom": 8,
          },
          "link": Object {
            "color": "#2389d7",
          },
          "mention": Object {
            "color": "#2389d7",
          },
          "mention_highlight": Object {
            "backgroundColor": "#ffe577",
          },
          "strong": Object {
            "fontWeight": "bold",
          },
          "table_header_row": Object {
            "fontWeight": "700",
          },
        }
      }
      value="Terms Text"
    />
  </ScrollViewMock>
</React.Fragment>
`;

exports[`TermsOfService should enable/disable navigator buttons on setNavigatorButtons true/false 2`] = `
<React.Fragment>
  <Connect(StatusBar) />
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
      }
    }
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "padding": 30,
      }
    }
  >
    <Connect(Markdown)
      baseTextStyle={
        Object {
          "color": "#3d3c40",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      blockStyles={
        Object {
          "adjacentParagraph": Object {
            "marginTop": 6,
          },
          "horizontalRule": Object {
            "backgroundColor": "#3d3c40",
            "flex": 1,
            "height": 0.5,
            "marginVertical": 10,
          },
          "quoteBlockIcon": Object {
            "color": undefined,
            "padding": 5,
          },
        }
      }
      disableAtMentions={true}
      disableChannelLink={true}
      disableHashtags={true}
      navigator={
        Object {
          "dismissAllModals": [MockFunction],
          "dismissModal": [MockFunction],
          "setButtons": [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": true,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": true,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": true,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": true,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": false,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": false,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": true,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": true,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "setOnNavigatorEvent": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        }
      }
      textStyles={
        Object {
          "code": Object {
            "alignSelf": "center",
            "backgroundColor": undefined,
            "fontFamily": "Menlo",
          },
          "codeBlock": Object {
            "fontFamily": "Menlo",
          },
          "del": Object {
            "textDecorationLine": "line-through",
          },
          "emph": Object {
            "fontStyle": "italic",
          },
          "error": Object {
            "color": "#fd5960",
          },
          "heading1": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading1Text": Object {
            "paddingBottom": 8,
          },
          "heading2": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading2Text": Object {
            "paddingBottom": 8,
          },
          "heading3": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading3Text": Object {
            "paddingBottom": 8,
          },
          "heading4": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading4Text": Object {
            "paddingBottom": 8,
          },
          "heading5": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading5Text": Object {
            "paddingBottom": 8,
          },
          "heading6": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading6Text": Object {
            "paddingBottom": 8,
          },
          "link": Object {
            "color": "#2389d7",
          },
          "mention": Object {
            "color": "#2389d7",
          },
          "mention_highlight": Object {
            "backgroundColor": "#ffe577",
          },
          "strong": Object {
            "fontWeight": "bold",
          },
          "table_header_row": Object {
            "fontWeight": "700",
          },
        }
      }
      value="Terms Text"
    />
  </ScrollViewMock>
</React.Fragment>
`;

exports[`TermsOfService should match snapshot 1`] = `
<Loading
  color="grey"
  size="large"
  style={Object {}}
/>
`;

exports[`TermsOfService should match snapshot for fail of get terms 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <FailedNetworkAction
    errorDescription={
      Object {
        "defaultMessage": "Make sure you have an active internet connection and try again. If this issue persists, contact your System Administrator.",
        "id": "mobile.terms_of_service.get_terms_error_description",
      }
    }
    errorTitle={
      Object {
        "defaultMessage": "Unable to load terms of service.",
        "id": "mobile.terms_of_service.get_terms_error_title",
      }
    }
    onRetry={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;

exports[`TermsOfService should match snapshot for get terms 1`] = `
<Loading
  color="grey"
  size="large"
  style={Object {}}
/>
`;

exports[`TermsOfService should match snapshot on enableNavigatorLogout 1`] = `
<React.Fragment>
  <Connect(StatusBar) />
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
      }
    }
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "padding": 30,
      }
    }
  >
    <Connect(Markdown)
      baseTextStyle={
        Object {
          "color": "#3d3c40",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      blockStyles={
        Object {
          "adjacentParagraph": Object {
            "marginTop": 6,
          },
          "horizontalRule": Object {
            "backgroundColor": "#3d3c40",
            "flex": 1,
            "height": 0.5,
            "marginVertical": 10,
          },
          "quoteBlockIcon": Object {
            "color": undefined,
            "padding": 5,
          },
        }
      }
      disableAtMentions={true}
      disableChannelLink={true}
      disableHashtags={true}
      navigator={
        Object {
          "dismissAllModals": [MockFunction],
          "dismissModal": [MockFunction],
          "setButtons": [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": true,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": true,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": true,
                      "icon": Object {},
                      "id": "reject-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": true,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
              Array [
                Object {
                  "leftButtons": Array [
                    Object {
                      "disabled": false,
                      "icon": Object {},
                      "id": "close-terms-of-service",
                    },
                  ],
                  "rightButtons": Array [
                    Object {
                      "disabled": true,
                      "id": "accept-terms-of-service",
                      "showAsAction": "always",
                      "title": undefined,
                    },
                  ],
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "setOnNavigatorEvent": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        }
      }
      textStyles={
        Object {
          "code": Object {
            "alignSelf": "center",
            "backgroundColor": undefined,
            "fontFamily": "Menlo",
          },
          "codeBlock": Object {
            "fontFamily": "Menlo",
          },
          "del": Object {
            "textDecorationLine": "line-through",
          },
          "emph": Object {
            "fontStyle": "italic",
          },
          "error": Object {
            "color": "#fd5960",
          },
          "heading1": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading1Text": Object {
            "paddingBottom": 8,
          },
          "heading2": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading2Text": Object {
            "paddingBottom": 8,
          },
          "heading3": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading3Text": Object {
            "paddingBottom": 8,
          },
          "heading4": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading4Text": Object {
            "paddingBottom": 8,
          },
          "heading5": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading5Text": Object {
            "paddingBottom": 8,
          },
          "heading6": Object {
            "fontSize": 17,
            "fontWeight": "700",
            "lineHeight": 25,
          },
          "heading6Text": Object {
            "paddingBottom": 8,
          },
          "link": Object {
            "color": "#2389d7",
          },
          "mention": Object {
            "color": "#2389d7",
          },
          "mention_highlight": Object {
            "backgroundColor": "#ffe577",
          },
          "strong": Object {
            "fontWeight": "bold",
          },
          "table_header_row": Object {
            "fontWeight": "700",
          },
        }
      }
      value="Terms Text"
    />
  </ScrollViewMock>
</React.Fragment>
`;
